internal/pkg/policy/policy_output.go

```diff
@@ -17,27 +17,34 @@ import (
 
 	"github.com/elastic/fleet-server/v7/internal/pkg/apikey"
 	"github.com/elastic/fleet-server/v7/internal/pkg/bulk"
+	"github.com/elastic/fleet-server/v7/internal/pkg/config"
 	"github.com/elastic/fleet-server/v7/internal/pkg/dl"
+	"github.com/elastic/fleet-server/v7/internal/pkg/es"
 	"github.com/elastic/fleet-server/v7/internal/pkg/logger"
 	"github.com/elastic/fleet-server/v7/internal/pkg/model"
 	"github.com/elastic/fleet-server/v7/internal/pkg/smap"
+	"github.com/elastic/go-elasticsearch/v8"
 )
 
 const (
-	OutputTypeElasticsearch = "elasticsearch"
-	OutputTypeLogstash      = "logstash"
-	OutputTypeKafka         = "kafka"
+	OutputTypeElasticsearch       = "elasticsearch"
+	OutputTypeRemoteElasticsearch = "remote_elasticsearch"
+	OutputTypeLogstash            = "logstash"
+	OutputTypeKafka               = "kafka"
 )
 
 var (
 	ErrNoOutputPerms    = errors.New("output permission sections not found")
 	ErrFailInjectAPIKey = errors.New("fail inject api key")
+	// TODO clean up bulkers when a remote output is removed
```

Comment:
We should not remove a bulker when the output is removed from the policy, that will lead to all the keys being orphaned. Our workflow now should be:
1. output is removed
2. fleet-server detects change and sends a new policy without (remote) output to agent
3. agent acks policy change
4. fleet-server revokes remote output api key.

It's not going to be safe to remove fleet-server's knowledge of a remote output before the api key is revoked and we'll likely need another way to track outputs in the future

Reply:
I am wondering if we should invalidate api key when the remote output is removed from a policy (and the agent acks), or only when the output is completely removed from Fleet?
Probably the latter, because the users can change the agent policy back to use the remote output as long as it exists.

Reply:
What I also don't understand, if I remove the remote ES output from an agent policy, the output is not removed from the agent output list in ES. So if I retire the remote API key and then change back the agent policy to the remote output, the agent will again try to use the old API key, which is invalidated.